aspnetcore/signalr/messagepackhubprotocol.md

```diff
@@ -0,0 +1,99 @@
+
+> [!NOTE]
+> JSON is enabled by default. Adding MessagePack enables support for both JSON and MessagePack clients.
+
+```csharp
+services.AddSignalR()
+    .AddMessagePackProtocol();
+```
+
+To customize how MessagePack will format your data, `AddMessagePackProtocol` takes a delegate for configuring options. In that delegate, the `FormatterResolvers` property can be used to configure MessagePack serialization options. For more information on how the resolvers work, visit the MessagePack library at [MessagePack-CSharp](https://github.com/neuecc/MessagePack-CSharp). Attributes can be used on the objects you want to serialize to define how they should be handled.
+
+```csharp
+services.AddSignalR()
+    .AddMessagePackProtocol(options =>
+    {
+        options.FormatterResolvers = new List<MessagePack.IFormatterResolver>()
+        {
+            MessagePack.Resolvers.StandardResolver.Instance
+        };
+    });
+```
+
+## Configure MessagePack on the client
+
+### .NET client
+
+To enable MessagePack in the .NET Client, install the `Microsoft.AspNetCore.SignalR.Protocols.MessagePack` package and call `AddMessagePackProtocol` on `HubConnectionBuilder`.
+
+```csharp
+var hubConnection = new HubConnectionBuilder()
+                        .WithUrl("/chatHub")
+                        .AddMessagePackProtocol()
+                        .Build();
+```
+
+> [!NOTE]
+> This `AddMessagePackProtocol` call takes a delegate for configuring options just like the server.
+
+### JavaScript client
+
+MessagePack support for the Javascript client is provided by the `@aspnet/signalr-protocol-msgpack` NPM package.
+
+```console
+npm install @aspnet/signalr-protocol-msgpack
+```
+
+After installing the npm package, the module can be used directly via a JavaScript module loader or imported into the browser by referencing the *node_modules\\@aspnet\signalr-protocol-msgpack\dist\browser\signalr-protocol-msgpack.js* file. When using the MessagePack library via a browser `<script>` tag. The msgpack5 library must be referenced directly as well, it can be found in the *node_modules\msgpack5\dist\msgpack5.js* file.
+
+> [!NOTE]
+> When using the `<script>` element, the order is important. If *signalr-protocol-msgpack.js* is referenced before *msgpack5.js*, an error occurs when trying to connect with MessagePack.
+
+```html
+<script src="~/lib/signalr/signalr.js"></script>
+<script src="~/lib/msgpack5/msgpack5.js"></script>
+<script src="~/lib/signalr/signalr-protocol-msgpack.js"></script>
+```
+
+Adding `.withHubProtocol(new signalR.protocols.msgpack.MessagePackHubProtocol())` to the `HubConnectionBuilder` will configure the client to use the MessagePack protocol when connecting to a server.
+
+```javascript
+const connection = new signalR.HubConnectionBuilder()
+    .withUrl("/chatHub")
+    .withHubProtocol(new signalR.protocols.msgpack.MessagePackHubProtocol())
+    .build();
+```
```

Comment:
Add a Note callout:

At this time, there are no configuration options for the MessagePack protocol on the JavaScript client.